### PR TITLE
Ts-sdk: change number to bigint or string in provider functions

### PR DESCRIPTION
**Pull request by danaiballa**
## Description 

- Function `getTotalTransactionNumber` of ts-sdk now returns a `bigint`. 
- Function `getReferenceGasPrice` of ts-sdk now returns a `bigint`.
- `gasPrice` of `devInspectTransaction` is defined as a `bigint`.
- `getLatestCheckpointSequenceNumber` returns a `string`
- `getCheckpoint` takes as input a `string` for checkpoint sequence number. 
- `getCheckpoints` takes as input a `string` for cursor.
- Introduce `SuiCheckpointSequenceNumber` type to sui-json-rpc-types that is defined as `BigInt` (compared to `CheckpointSequenceNumber` used from sui-types that was defined as a `u64`)
- Change rust side to support the above. Changes to explorer, sui-sdk, sui-indexer and sui-rosetta that got affected by the changes.

PR #9347 was adding a lot of changes, so I am splitting it into multiple PRs so that the changes can be better reviewed and tested. A related PR: #9462 

## Test Plan 

manually test explorer and wallet

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

**Comment by danaiballa** [comment]
For `epoch` inputs such as in `devInspectTransaction` and `getCommitteeInfo`, `epoch` could remain a number since it may not get a value that cannot fit into a `number` type.
For `getLatestCheckpointSequenceNumber`, the value that is returned by the sui-json-rpc is `CheckpointSequenceNumber` and is defined in sui-types. Changing it to a `BigInt` so that `getLatestCheckpointSequenceNumber` can return a `bigint` may cause a lot of things to break.

File: sdk/typescript/src/providers/json-rpc-provider.ts [hunk 1]
```diff
@@ -1016,15 +1018,15 @@ export class JsonRpcProvider {
   /**
    * Get the sequence number of the latest checkpoint that has been executed
    */
-  async getLatestCheckpointSequenceNumber(): Promise<number> {
+  async getLatestCheckpointSequenceNumber(): Promise<bigint> {
```
Comment (Jordan-Mysten) [inline sdk/typescript/src/providers/json-rpc-provider.ts:1021]: I'd also be totally fine if these just returned strings.

**Review by 666lcz — APPROVED** [review]

**Comment by 666lcz** [comment]
Please rebase + do some final testing on explorer/wallet before merging

**Review by kchalkias — COMMENTED** [review]
As Chris mentioned, we always prefer rebasing vs merging to main when submitting PRs, please rebase.

**Pull request merged**